azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java

```diff
@@ -467,7 +533,7 @@
          * @param name the name for the data disk
          * @return the stage representing configuration for the data disk
          */
-        DataDisk.DefinitionAttachNewDataDisk<DefinitionCreatable> defineNewDataDisk(String name);
```

Comment:
in the impl, we should be almost always able to return the impl type itself. The interfaces take care of presenting the right interface return type, it shouldn't be the impl's job. See NetworkImpl#defineSubnet() as an example of what I mean.